language/move-core/types/src/u256.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use ethnum::U256 as EthnumU256;
+use num::{bigint::Sign, BigInt};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::strategy::BoxedStrategy;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::strategy::Strategy;
+use std::{
+    fmt,
+    mem::size_of,
+    ops::{BitAnd, BitOr, BitXor, Shl, Shr},
+};
+use uint::FromStrRadixErr;
+
+const NUM_BITS_PER_BYTE: usize = 8;
+const U256_NUM_BITS: usize = 256;
+const U256_NUM_BYTES: usize = U256_NUM_BITS / NUM_BITS_PER_BYTE;
+pub type U256FromStrError = FromStrRadixErr;
+
+// This U256 impl was chosen for now but we are open to changing it as needed
+use primitive_types::U256 as PrimitiveU256;
+use serde::{ser::SerializeTuple, Deserialize, Deserializer, Serialize, Serializer};
+
+#[allow(non_camel_case_types)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Copy, PartialOrd, Ord)]
+pub struct U256(PrimitiveU256);
+
+impl fmt::Display for U256 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::str::FromStr for U256 {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        PrimitiveU256::from_str(s).map(Self).map_err(|e| e.into())
+    }
+}
+
+impl<'de> Deserialize<'de> for U256 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(U256::from_le_bytes(
+            &(<[u8; U256_NUM_BYTES]>::deserialize(deserializer)?),
+        ))
+    }
+}
+
+impl Serialize for U256 {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let b: [u8; U256_NUM_BYTES] = self.to_le_bytes();
+        let mut seq = serializer.serialize_tuple(U256_NUM_BYTES)?;
+        for x in b {
+            seq.serialize_element(&x)?;
+        }
+        seq.end()
```

Comment:
[u8; 32] does implement Serialize so you may be able to do
```suggestion
        self.to_le_bytes().serialize(serializer)
```